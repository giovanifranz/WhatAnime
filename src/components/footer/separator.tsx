'use client'

import { useWindowSize } from 'usehooks-ts'

import { Separator as UiSeparator } from '../ui/separator'

export function Separator() {
  const { width } = useWindowSize()

  return (
    <UiSeparator
      orientation={width < 768 ? 'horizontal' : 'vertical'}
      className="my-auto bg-neutral-200 md:my-4 md:flex md:h-72"
    />
  )
}
