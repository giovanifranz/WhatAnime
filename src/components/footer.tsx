import Image from 'next/image'
import Link from 'next/link'
import {
  RiInstagramFill,
  RiFacebookFill,
  RiTwitterFill,
  RiPaypalFill,
} from 'react-icons/ri'
import { SiAnilist, SiMyanimelist } from 'react-icons/si'

import Highlander from '@/assets/highlander.svg'

import { Logo } from './logo'
import { Social } from './social'
import { Separator } from './ui/separator'

export function Footer() {
  return (
    <footer className="bg-neutral-800">
      <div className="mx-auto flex max-w-7xl gap-4 p-4">
        <div className="w-2/5">
          <Logo />
          <div className="mt-8 flex flex-col gap-2">
            <span>CONNECT</span>
            <div className="flex gap-4">
              <Social href="/" icon={SiMyanimelist} />
              <Social href="/" icon={SiAnilist} />
            </div>

            <span>SHARE</span>
            <div className="flex gap-4">
              <Social href="/" icon={RiFacebookFill} />
              <Social href="/" icon={RiInstagramFill} />
              <Social href="/" icon={RiTwitterFill} />
            </div>
          </div>
        </div>
        <Separator orientation="vertical" className="my-auto h-72 bg-neutral-200" />
        <div className="flex flex-col gap-4">
          <div className="flex items-center gap-4">
            <p className="max-w-xs">
              This site is our baby, so please consider donating if you are able so that
              we can keep working on it!
            </p>
            <Link href="/" className="rounded-full bg-neutral-100 p-2 text-neutral-800">
              <RiPaypalFill size={56} />
            </Link>
          </div>
          <span className="mt-4">LEGAL</span>
          <p>
            All anime series names, images, and content are copyrighted content of their
            respective license holders. I do not own the rights to any of these anime
            series. Anime information compiled from AniList and MyAnimeList.
          </p>
          <span className="-mb-4 mt-4">DESIGNED BY</span>
          <Link href="https://www.highlandertech.com.br/">
            <Image src={Highlander} alt="HighlanderTech" width={257} height={65} />
          </Link>
        </div>
      </div>
    </footer>
  )
}
