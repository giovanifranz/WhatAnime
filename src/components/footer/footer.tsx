import Image from 'next/image'
import Link from 'next/link'
import { RiFacebookFill, RiTwitterFill, RiPaypalFill } from 'react-icons/ri'
import { SiAnilist, SiMyanimelist } from 'react-icons/si'

import Highlander from '@/assets/highlander.svg'

import { Logo } from '@/components'

import { Separator } from './separator'
import { Social } from './social'

export function Footer() {
  return (
    <footer className="bg-neutral-800">
      <div className="mx-auto flex flex-col gap-4 p-4 md:max-w-7xl md:flex-row">
        <div className="w-full md:w-2/5">
          <Logo />
          <div className="mt-8 flex flex-col gap-2">
            <span>CONNECT</span>
            <div className="flex gap-4">
              <Social
                data-testid="myanimelist-social"
                href="https://myanimelist.net/profile/HighlanderTech"
                icon={SiMyanimelist}
              />
              <Social
                data-testid="anilist-social"
                href="https://anilist.co/user/giovanifranz/"
                icon={SiAnilist}
              />
            </div>

            <span>SHARE</span>
            <div className="flex gap-4">
              <Social
                data-testid="facebook-social"
                href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fwww.whatanime.com.br"
                icon={RiFacebookFill}
              />
              <Social
                data-testid="twitter-social"
                href="https://twitter.com/intent/tweet?url=https%3A%2F%2Fwww.whatanime.com.br"
                icon={RiTwitterFill}
              />
            </div>
          </div>
        </div>

        <Separator />

        <div className="flex flex-col gap-4">
          <div className="flex items-center gap-4">
            <p className="max-w-xs">
              This site is our baby, so please consider donating if you are able so that
              we can keep working on it!
            </p>
            <Link
              data-testid="paypal"
              href="https://www.paypal.com/donate/?hosted_button_id=KJ9TK628E7N42"
              className="rounded-full bg-neutral-100 p-2 text-neutral-800 transition-all hover:opacity-90"
            >
              <RiPaypalFill size={56} />
            </Link>
          </div>
          <span className="mt-4">LEGAL</span>
          <p>
            All anime series names, images, and content are copyrighted content of their
            respective license holders. I do not own the rights to any of these anime
            series. Anime information compiled from AniList and MyAnimeList.
          </p>
          <span className="-mb-4 mt-4">DESIGNED BY</span>
          <Link href="https://www.highlandertech.com.br" data-testid="highlander-logo">
            <Image src={Highlander} alt="HighlanderTech" width={257} height={65} />
          </Link>
        </div>
      </div>
    </footer>
  )
}
