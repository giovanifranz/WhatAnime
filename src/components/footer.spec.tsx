import { render } from '@/test/test-utils'

import { Footer } from './footer'

describe('Footer', () => {
  it('Deve renderizar o Footer corretamente', () => {
    const { asFragment } = render(<Footer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
