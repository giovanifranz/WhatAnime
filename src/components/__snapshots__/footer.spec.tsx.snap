// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > Deve renderizar o Footer corretamente 1`] = `
<DocumentFragment>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('dark')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'dark'}catch(e){}}()
  </script>
  <footer
    class="bg-neutral-800"
  >
    <div
      class="mx-auto flex max-w-7xl gap-4 p-4"
    >
      <div
        class="w-2/5"
      >
        <a
          href="/"
        >
          <img
            alt="WhatAnime"
            data-nimg="1"
            decoding="async"
            height="76"
            loading="lazy"
            src="/src/assets/logo.svg"
            style="color: transparent;"
            width="227"
          />
        </a>
        <div
          class="mt-8 flex flex-col gap-2"
        >
          <span>
            CONNECT
          </span>
          <div
            class="flex gap-4"
          >
            <a
              class="rounded-lg bg-neutral-50 p-2 text-neutral-800 transition-all hover:opacity-90"
              href="/"
            >
              <svg
                fill="currentColor"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title />
                <path
                  d="M8.273 7.247v8.423l-2.103-.003v-5.216l-2.03 2.404-1.989-2.458-.02 5.285H.001L0 7.247h2.203l1.865 2.545 2.015-2.546 2.19.001zm8.628 2.069l.025 6.335h-2.365l-.008-2.871h-2.8c.07.499.21 1.266.417 1.779.155.381.298.751.583 1.128l-1.705 1.125c-.349-.636-.622-1.337-.878-2.082a9.296 9.296 0 0 1-.507-2.179c-.085-.75-.097-1.471.107-2.212a3.908 3.908 0 0 1 1.161-1.866c.313-.293.749-.5 1.1-.687.351-.187.743-.264 1.107-.359a7.405 7.405 0 0 1 1.191-.183c.398-.034 1.107-.066 2.39-.028l.545 1.749H14.51c-.593.008-.878.001-1.341.209a2.236 2.236 0 0 0-1.278 1.92l2.663.033.038-1.81h2.309zm3.992-2.099v6.627l3.107.032-.43 1.775h-4.807V7.187l2.13.03z"
                />
              </svg>
            </a>
            <a
              class="rounded-lg bg-neutral-50 p-2 text-neutral-800 transition-all hover:opacity-90"
              href="/"
            >
              <svg
                fill="currentColor"
                height="24"
                role="img"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title />
                <path
                  d="M6.361 2.943L0 21.056h4.942l1.077-3.133H11.4l1.052 3.133H22.9c.71 0 1.1-.392 1.1-1.101V17.53c0-.71-.39-1.101-1.1-1.101h-6.483V4.045c0-.71-.392-1.102-1.101-1.102h-2.422c-.71 0-1.101.392-1.101 1.102v1.064l-.758-2.166zm2.324 5.948l1.688 5.018H7.144z"
                />
              </svg>
            </a>
          </div>
          <span>
            SHARE
          </span>
          <div
            class="flex gap-4"
          >
            <a
              class="rounded-lg bg-neutral-50 p-2 text-neutral-800 transition-all hover:opacity-90"
              href="/"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 13.5H16.5L17.5 9.5H14V7.5C14 6.47062 14 5.5 16 5.5H17.5V2.1401C17.1743 2.09685 15.943 2 14.6429 2C11.9284 2 10 3.65686 10 6.69971V9.5H7V13.5H10V22H14V13.5Z"
                />
              </svg>
            </a>
            <a
              class="rounded-lg bg-neutral-50 p-2 text-neutral-800 transition-all hover:opacity-90"
              href="/"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.0281 2.00098C14.1535 2.00284 14.7238 2.00879 15.2166 2.02346L15.4107 2.02981C15.6349 2.03778 15.8561 2.04778 16.1228 2.06028C17.1869 2.10944 17.9128 2.27778 18.5503 2.52528C19.2094 2.77944 19.7661 3.12278 20.3219 3.67861C20.8769 4.23444 21.2203 4.79278 21.4753 5.45028C21.7219 6.08694 21.8903 6.81361 21.9403 7.87778C21.9522 8.14444 21.9618 8.36564 21.9697 8.58989L21.976 8.78397C21.9906 9.27672 21.9973 9.8471 21.9994 10.9725L22.0002 11.7182C22.0003 11.8093 22.0003 11.9033 22.0003 12.0003L22.0002 12.2824L21.9996 13.0281C21.9977 14.1535 21.9918 14.7238 21.9771 15.2166L21.9707 15.4107C21.9628 15.6349 21.9528 15.8561 21.9403 16.1228C21.8911 17.1869 21.7219 17.9128 21.4753 18.5503C21.2211 19.2094 20.8769 19.7661 20.3219 20.3219C19.7661 20.8769 19.2069 21.2203 18.5503 21.4753C17.9128 21.7219 17.1869 21.8903 16.1228 21.9403C15.8561 21.9522 15.6349 21.9618 15.4107 21.9697L15.2166 21.976C14.7238 21.9906 14.1535 21.9973 13.0281 21.9994L12.2824 22.0002C12.1913 22.0003 12.0973 22.0003 12.0003 22.0003L11.7182 22.0002L10.9725 21.9996C9.8471 21.9977 9.27672 21.9918 8.78397 21.9771L8.58989 21.9707C8.36564 21.9628 8.14444 21.9528 7.87778 21.9403C6.81361 21.8911 6.08861 21.7219 5.45028 21.4753C4.79194 21.2211 4.23444 20.8769 3.67861 20.3219C3.12278 19.7661 2.78028 19.2069 2.52528 18.5503C2.27778 17.9128 2.11028 17.1869 2.06028 16.1228C2.0484 15.8561 2.03871 15.6349 2.03086 15.4107L2.02457 15.2166C2.00994 14.7238 2.00327 14.1535 2.00111 13.0281L2.00098 10.9725C2.00284 9.8471 2.00879 9.27672 2.02346 8.78397L2.02981 8.58989C2.03778 8.36564 2.04778 8.14444 2.06028 7.87778C2.10944 6.81278 2.27778 6.08778 2.52528 5.45028C2.77944 4.79194 3.12278 4.23444 3.67861 3.67861C4.23444 3.12278 4.79278 2.78028 5.45028 2.52528C6.08778 2.27778 6.81278 2.11028 7.87778 2.06028C8.14444 2.0484 8.36564 2.03871 8.58989 2.03086L8.78397 2.02457C9.27672 2.00994 9.8471 2.00327 10.9725 2.00111L13.0281 2.00098ZM12.0003 7.00028C9.23738 7.00028 7.00028 9.23981 7.00028 12.0003C7.00028 14.7632 9.23981 17.0003 12.0003 17.0003C14.7632 17.0003 17.0003 14.7607 17.0003 12.0003C17.0003 9.23738 14.7607 7.00028 12.0003 7.00028ZM12.0003 9.00028C13.6572 9.00028 15.0003 10.3429 15.0003 12.0003C15.0003 13.6572 13.6576 15.0003 12.0003 15.0003C10.3434 15.0003 9.00028 13.6576 9.00028 12.0003C9.00028 10.3434 10.3429 9.00028 12.0003 9.00028ZM17.2503 5.50028C16.561 5.50028 16.0003 6.06018 16.0003 6.74943C16.0003 7.43867 16.5602 7.99944 17.2503 7.99944C17.9395 7.99944 18.5003 7.43954 18.5003 6.74943C18.5003 6.06018 17.9386 5.49941 17.2503 5.50028Z"
                />
              </svg>
            </a>
            <a
              class="rounded-lg bg-neutral-50 p-2 text-neutral-800 transition-all hover:opacity-90"
              href="/"
            >
              <svg
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22.2125 5.65605C21.4491 5.99375 20.6395 6.21555 19.8106 6.31411C20.6839 5.79132 21.3374 4.9689 21.6493 4.00005C20.8287 4.48761 19.9305 4.83077 18.9938 5.01461C18.2031 4.17106 17.098 3.69303 15.9418 3.69434C13.6326 3.69434 11.7597 5.56661 11.7597 7.87683C11.7597 8.20458 11.7973 8.52242 11.8676 8.82909C8.39047 8.65404 5.31007 6.99005 3.24678 4.45941C2.87529 5.09767 2.68005 5.82318 2.68104 6.56167C2.68104 8.01259 3.4196 9.29324 4.54149 10.043C3.87737 10.022 3.22788 9.84264 2.64718 9.51973C2.64654 9.5373 2.64654 9.55487 2.64654 9.57148C2.64654 11.5984 4.08819 13.2892 6.00199 13.6731C5.6428 13.7703 5.27232 13.8194 4.90022 13.8191C4.62997 13.8191 4.36771 13.7942 4.11279 13.7453C4.64531 15.4065 6.18886 16.6159 8.0196 16.6491C6.53813 17.8118 4.70869 18.4426 2.82543 18.4399C2.49212 18.4402 2.15909 18.4205 1.82812 18.3811C3.74004 19.6102 5.96552 20.2625 8.23842 20.2601C15.9316 20.2601 20.138 13.8875 20.138 8.36111C20.138 8.1803 20.1336 7.99886 20.1256 7.81997C20.9443 7.22845 21.651 6.49567 22.2125 5.65605Z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="shrink-0 w-[1px] my-auto h-72 bg-neutral-200"
        data-orientation="vertical"
        role="none"
      />
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="flex items-center gap-4"
        >
          <p
            class="max-w-xs"
          >
            This site is our baby, so please consider donating if you are able so that we can keep working on it!
          </p>
          <a
            class="rounded-full bg-neutral-100 p-2 text-neutral-800"
            href="/"
          >
            <svg
              fill="currentColor"
              height="56"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="56"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.0673 8.47768C20.5591 9.35823 20.6237 10.4924 20.3676 11.8053C19.627 15.6107 17.0916 16.9253 13.8536 16.9253H13.3536C12.9583 16.9253 12.6216 17.214 12.5596 17.6047L12.519 17.8253L11.8896 21.818L11.857 21.988C11.795 22.3787 11.4583 22.6667 11.063 22.6667H7.72031C7.42365 22.6667 7.19698 22.402 7.24298 22.1093L7.41807 21H8.9367L9.88603 14.9793H11.2716C15.9496 14.9793 19.0209 12.7768 20.0673 8.47768ZM17.1066 3.38784C17.8693 4.25635 18.0908 5.19891 17.8597 6.67324C17.8405 6.79594 17.82 6.91391 17.7973 7.03253C17.0621 10.8057 14.7087 12.4793 10.8417 12.4793H8.95703C8.32647 12.4793 7.78368 12.8928 7.60372 13.4811L7.58913 13.4788L6.65969 19.3733H3.12169C3.08991 19.3733 3.06598 19.3454 3.07097 19.3136L5.66905 2.80233C5.74174 2.34036 6.13984 2 6.6075 2H12.583C14.7658 2 16.2998 2.46869 17.1066 3.38784Z"
              />
            </svg>
          </a>
        </div>
        <span
          class="mt-4"
        >
          LEGAL
        </span>
        <p>
          All anime series names, images, and content are copyrighted content of their respective license holders. I do not own the rights to any of these anime series. Anime information compiled from AniList and MyAnimeList.
        </p>
        <span
          class="-mb-4 mt-4"
        >
          DESIGNED BY
        </span>
        <a
          href="https://www.highlandertech.com.br/"
        >
          <img
            alt="HighlanderTech"
            data-nimg="1"
            decoding="async"
            height="65"
            loading="lazy"
            src="/src/assets/highlander.svg"
            style="color: transparent;"
            width="257"
          />
        </a>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
